### PR TITLE
Design: hardware that cannot be powered off

### PR DESCRIPTION
**Pull request by dtantsur**
Add a new design proposal to support hardware that must never be powered
off completely (targeting primarily NC-SI architectures).

Signed-off-by: Dmitry Tantsur <dtantsur@protonmail.com>

### CODE REVIEW

**Review by hroyrh — COMMENTED** [review]
Looks good to me.

File: design/baremetal-operator/disable-power-off.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+### Non-Goals
+
+- Changing the default behavior.
+
+## Proposal
+
+## Design Details
+
+Add a new field `DisablePowerOff` (boolean, default `false`) to the
```
[the remark this answers is not in the document]
Comment (dtantsur) [reply design/baremetal-operator/disable-power-off.md:50]: I'm struggling with the difference myself. But with my API purist hat on, API fields are easy to discover and to verify. I honestly think we're overdoing it with annotations in BMO a bit: disabling inspection could be a field too.

**Review by Rozzii — APPROVED** [review]
/hold
/approve
I have put the hold on this just to give the opportunity for other to also review, also there is one unanswered comment.

In my POV this feature is  a good improvement.

File: design/baremetal-operator/disable-power-off.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+The `PoweringOffBeforeDelete` state will be skipped for hosts with
+`DisablePowerOff` set to `true`.
+
+Check the [Ironic specification][ironic-ncsi-spec] for more implementation
+details.
+
+### Risks and Mitigations
+
+The code paths without power off will be less tested than the normal path and
```
[the remark this answers is not in the document]
Comment (dtantsur) [reply design/baremetal-operator/disable-power-off.md:73]: I'm not sure it makes the matters better: your sounds like the power off operation is not available at all.

[the remark this answers is not in the document]
Comment (hroyrh) [reply design/baremetal-operator/disable-power-off.md:73]: Hmm, right it does look like that, maybe something like - "The new code paths", but I'm fine either way ( reading it again, it looks good to me ).

**Comment by dtantsur** [comment]
/hold cancel

Looks like we have consensus.

**Comment by dtantsur** [comment]
@kashifest @Rozzii @elfosardo could someone please re-add lgtm? I had to make the spell checker happy (no other changes).

**Pull request merged**